### ISSUE TITLE
Why is V2's FPS slower than V1's?

### ISSUE THREAD

**Issue by Kyrie10favor** [CoinCheung/BiSeNet#158]
hello,CoinCheung:
  Why is V2's FPS slower than V1's?
  thx.

**Comment by CoinCheung** [comment]
Because bisenetv2 uses many depth-width convolution, which requires accessing gpu global memory sparsely. This would bring more memory latency than continuous operations such as fully-connected or convolutions. Slower model does not means the model requires more computation, the overhead can be brought by other conditions.

**Comment by Kyrie10favor** [comment]
Thanks for your reply,
And how do I get the right FPS in the inference?
Here's what I did:
`
                a = time.time()
                **logits = net(im_sc)[0]**
                b = time.time()
                fps[1]+=(b-a)   # Total time-consuming
                fps[0]+=1   #Total  inferences
`
and then fps=fps[0] / fps[1]
Am I doing the right thing?

**Comment by CoinCheung** [comment]
I tested that with tensorrt. That is what the authors reported in the paper. You can refere to this: https://github.com/CoinCheung/BiSeNet/tree/master/tensorrt

If you would like to test fps with python, you can do with the method you posted. Please pay attention to two things: 1). some cuda operations are not synchronized, so you need to add a `torch.cuda.synchronize()` before recording the end time; 2). some model contains auxiliary segment head which is useful for training but not for inference, you need to remove them to test your fps.

**Comment by Kyrie10favor** [comment]
Okay. Got it.

**Comment by CoinCheung** [comment]
I am closing this, since the problem seems to be solved now.
